src/TraceEvent/Computers/TraceManagedProcess.cs

```diff
@@ -1010,6 +1010,59 @@ internal static void SetupCallbacks(TraceEventDispatcher source)
                             finalizationCount + 1 :
                             1;
                 };
+
+                source.Clr.TypeBulkType += data =>
+                {
+                    var stats = currentManagedProcess(data);
+                    var typeMapping = stats.GC.m_stats.TypeMapping;
+
+                    var clonedData = data;
+                    clonedData = (GCBulkTypeTraceData)clonedData.Clone();
+
+                    var count = clonedData.Count;
+                    for (int i = 0; i < count; i++)
+                    {
+                        var bulkTypeValues = clonedData.Values(i);
+                        var typeId = bulkTypeValues.TypeID;
+                        if (typeMapping.ContainsKey(typeId))
+                            continue;
+
+                        typeMapping[typeId] = bulkTypeValues;
+                    }
+                };
+
+                source.Clr.GCFinalizeObject += data =>
```

Comment:
❓ @vancem What is the relation between this added line and [this line](https://github.com/Microsoft/perfview/blob/06fb35cf868f32a730ed5b74c4e426d76f6475ee/src/TraceEvent/TraceLog.cs#L880)?